src/Analyser/NodeScopeResolver.php

```diff
@@ -3295,7 +3310,11 @@ private function processArgs(
 				if ($assignByReference) {
 					$argValue = $arg->value;
 					if ($argValue instanceof Variable && is_string($argValue->name)) {
-						$scope = $scope->assignVariable($argValue->name, new MixedType(), new MixedType());
+						$byRefType = new MixedType();
+						if (isset($parameters[$i]) && isset($paramOutTypes[$parameters[$i]->getName()])) {
+							$byRefType = $paramOutTypes[$parameters[$i]->getName()];
```

Comment:
I'd do it a few lines above in the if/elseif to account for this edge-case: https://3v4l.org/QURqf (and with a test of course)

Reply:
I don't think so, `@param-out int $var` is fine

Reply:
The `...` information is already in the native signature and/or in `@param`.

Reply:
I see, thanks. added a test and the proposed fix.